source/module_base/cubic_spline.cpp

```diff
@@ -124,13 +125,18 @@ void CubicSpline::build(const int n,
     }
     else
     {
-        double* dx = new double[n - 1];
```

Comment:
I remember you used to refactor this part, I wonder way you did not use std::vector then? Do you have some special consideration? About heterogeneous calculation? @jinzx10